src/hotspot/share/oops/constMethodFlags.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_OOPS_CONSTMETHODFLAGS_HPP
+#define SHARE_OOPS_CONSTMETHODFLAGS_HPP
+
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/macros.hpp"
+
+class outputStream;
+
+// The ConstMethodFlags class contains the parse-time flags associated with
+// an Method, and their associated accessors.
+// These flags are JVM internal and not part of the AccessFlags classfile specification.
+
+class ConstMethodFlags {
+  friend class VMStructs;
+  friend class JVMCIVMStructs;
+
+#define CM_FLAGS_DO(flag)  \
+   flag(has_linenumber_table      , 1 << 0) \
+   flag(has_checked_exceptions    , 1 << 1) \
+   flag(has_localvariable_table   , 1 << 2) \
+   flag(has_exception_table       , 1 << 3) \
+   flag(has_generic_signature     , 1 << 4) \
+   flag(has_method_parameters     , 1 << 5) \
+   flag(is_overpass               , 1 << 6) \
+   flag(has_method_annotations    , 1 << 7) \
+   flag(has_parameter_annotations , 1 << 8) \
+   flag(has_type_annotations      , 1 << 9) \
+   flag(has_default_annotations   , 1 << 10) \
+   flag(caller_sensitive          , 1 << 11) \
```

Comment:
Nit: we should consistently use either `x` or `is_x` for `x` in `overpass, caller_sensitive, hidden, scoped, ...`

Reply:
That's my preference too, but I was trying to not change all the callers to is_caller_sensitive, for example, or providing a set of wrappers to change the "x" calls to "is_x" calls to the flags interface.

Reply:
The 'is' in not_x_compilable can be added without too much fanout of lines and files that I don't want to change, so I added the 'is' to these.